src/custom/hooks/useRecentActivity.ts

```diff
@@ -0,0 +1,93 @@
+import { useMemo } from 'react'
+import { isTransactionRecent, useAllTransactions } from 'state/transactions/hooks'
+import { useOrders } from 'state/orders/hooks'
+import { useActiveWeb3React } from 'hooks'
+import { Order, OrderStatus } from 'state/orders/actions'
+import { TransactionDetails } from '@src/state/transactions/reducer'
+
+type TransactionAndOrder =
+  | (Order & { addedTime: number })
+  | (TransactionDetails & {
+      id: string
+      status: OrderStatus
+    })
+
+// One fill day in MS
+const DAY_MS = 86_400_000
+
+/**
+ * Returns whether a order happened in the last day (86400 seconds * 1000 milliseconds / second)
+ * @param order
+ */
+export function isOrderRecent(order: Order): boolean {
+  return Date.now() - Date.parse(order.creationTime) < DAY_MS
+}
+
+function isPending(data: TransactionAndOrder) {
+  return data.status === OrderStatus.PENDING
+}
+
+function isFulfilled(data: TransactionAndOrder) {
+  return data.status === OrderStatus.FULFILLED
+}
+
+/**
+ * useRecentActivity
+ * @description returns all RECENT (last day) transaction and orders in 2 arrays: pending and confirmed
+ */
+export default function useRecentActivity() {
+  const { chainId } = useActiveWeb3React()
+  const allTransactions = useAllTransactions()
+  const allNonEmptyOrders = useOrders({ chainId })
+
+  const recentOrdersAdjusted = useMemo<TransactionAndOrder[]>(() => {
+    // Filter out any pending/fulfilled orders OLDER than 1 day
+    // and adjust order object to match TransactionDetail addedTime format
+    // which is used later in app to render list of activity
+    const adjustedOrders = allNonEmptyOrders.filter(isOrderRecent).map(order => {
+      // we need to essentially match TransactionDetails type which uses "addedTime" for date checking
+      // and time in MS vs ISO string as Orders uses
+      return {
+        ...order,
+        addedTime: Date.parse(order.creationTime)
```

Comment:
Wait, `Order.creationTime` is unix epoch in seconds, we should `* 1000`
Correct me @anxolin

Reply:
let me ask backend bros

Reply:
```
Newest api says:
creationTime:
          description: Creation time of the order. Encoded as ISO 8601 UTC.
          type: string
          example: "2020-12-03T18:35:18.814523Z"
But I think that many times we are just sending:
"2020-12-03T18:35:18.81Z"
Does this makes a difference for you?
```

Date.parse will give us MS

Reply:
Sounds good

Reply:
Yes @W3stside  u should be able to check swagger https://protocol-rinkeby.dev.gnosisdev.com/api

Reply:
:+1: